### PR TITLE
[Display_01]Build DisplayGpsReceiverImplementationTests

### PR DESCRIPTION
**Pull request by EC01EC**
Working on #67

### CODE REVIEW

File: src/test/java/edu/oakland/test/Display01/DisplayGpsReceiverImplementation/DisplayGpsReceiverImplementationTests.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import edu.oakland.production.display01.DisplayGpsReceiverImplementation;
+import java.lang.IllegalArgumentException;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+
+
+@DisplayName ("Display GPS Receiver Implementation")
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/Display01/DisplayGpsReceiverImplementation/DisplayGpsReceiverImplementationTests.java]: For sake of standardization to make it easier to read and search the logs, can we change this to `DisplayGpsReceiverImplementation Unit Tests`? Thank you!
```suggestion
@DisplayName ("DisplayGpsReceiverImplementation Unit Tests")
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java [hunk 2]
```diff
@@ -1,20 +1,35 @@
...
    *
    * @param satelliteSignal GPS signal to measure.
-   * @return Status of GPS signal measured.
    */
   public String measureGpsSignalStrength(Satellite satelliteSignal) {
-
-    return "";
-
+    if (satelliteSignal == null) {
+      throw new IllegalArgumentException("Signal strength cannot be null");
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java]: Just to make this exception a little more descriptive, can we change the output to the name of the parameter? Thank you!
```suggestion
      throw new IllegalArgumentException("satelliteSignal cannot be null");
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java [hunk 3]
```diff
@@ -24,9 +39,9 @@ public String measureGpsSignalStrength(Satellite satelliteSignal) {
    * @return Request to reconnect to satellite.
    */
   public SatelliteSignalCheckRequest measureSignal(Satellite satelliteSignal) {
-        
-    return null;
-
+    if (satelliteSignal == null) {
+      throw new IllegalArgumentException("Signal strength cannot be null");
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java]: Just to make this exception a little more descriptive, can we change the output to the name of the parameter? Thank you!
```suggestion
      throw new IllegalArgumentException("satelliteSignal cannot be null");
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Great start so far! The production code looks good just in terms of a quick read though, we just need to make sure we've got test coverage for it. Based on the code that we have here, there are 4 more tests that you'll need to cover the rest of the production code:

- [x] measureGpsSignalStrengthSatelliteSignalNotNull()
- [x] measureGpsSignalStrengthSignalInIsSignalOut()
- [x] measureSignalSatelliteSignalNotNull()
- [x] measureSignalSignalInIsSignalOut()

To complete these tests, you'll create a new class in the `edu.oakland.test.display01` package called `DisplayGisManagerStub` that implements the `DisplayGisManager` Java interface. For each method defined in the `DisplayGisManagerStub` based on the interface, you'll create and return a new `SatelliteSignalCheckRequest` object with the name of the input satellite as the name of the satellite in the aforementioned output object.

From there, you can simply add the methods for the 4 test cases above.

For each of the "Not Null" cases create a new instance of `DisplayGpsReceiverImplementation` with an instance of `DisplayGisManagerStub` passed in via the constructor, then call the appropriate method based on the test name from within the `assertThrows` test as you currently do for `displayGpsManagerIsNotNull()`. 

For each of the "In Is Out" cases create a new instance of `DisplayGpsReceiverImplementation` with an instance of `DisplayGisManagerStub` passed in via the constructor. Then, you'll create a Satellite object with a non-empty name, and call the appropriate method based on the test name from within the `assertEquals` test to verify that `inputSatellite.getSatelliteName()` is equal to `displayGpsReceiver.methodHere(inputSatellite).getSatelliteName()`. 

If you have any issues or questions, feel free to reach out!

**Review by stevecatherman — APPROVED** [review]
Great success!

File: src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java [hunk 4]
```diff
@@ -1,20 +1,36 @@
...
+  private DisplayGpsManager displayGpsManager;
+
+  /**
+   * Checks whether or not passed displayGpsManager value is null.
+   *
+   * @param displayGpsManager Calls on displayGpsManager to check value for null.
+   */
+  public DisplayGpsReceiverImplementation(DisplayGpsManager displayGpsManager) {
+    if (displayGpsManager == null) {
+      throw new IllegalArgumentException("Signal strength cannot be null");
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java]: This is actually checking the manager, not the signal strength.
```suggestion
      throw new IllegalArgumentException("displayGpsManager cannot be null");
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsManagerStub.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  
+  /**
+   * Receives Gps Signal Strength and pulls satellite name.
+   *
+   * @param satelliteSignal passes satellite object to pull name and checktype from
+   */
+  public String receiveGpsSignalStrength(Satellite satelliteSignal) {
+    SatelliteSignalCheckRequest ssrc = 
+        new SatelliteSignalCheckRequest(satelliteSignal.getSatelliteName(), checkType);
+    return ssrc.getSatelliteName();
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsManagerStub.java]: This just needs to return a string, so we can streamline it as follows:
```suggestion
    return satelliteSignal.getSatelliteName();
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsManagerStub.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  /**
+   * Receives Gps Signal Strength and pulls satellite name.
+   *
+   * @param satelliteSignal passes satellite object to pull name and checktype from
+   */
+  public SatelliteSignalCheckRequest passGpsSignalStrength(Satellite satelliteSignal) {
+    SatelliteSignalCheckRequest ssrc = 
+        new SatelliteSignalCheckRequest(satelliteSignal.getSatelliteName(), checkType);
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsManagerStub.java]: Just a little cleaner formatting:
```suggestion
    SatelliteSignalCheckRequest ssrc = new SatelliteSignalCheckRequest(
        satelliteSignal.getSatelliteName(),
        checkType
    );
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    assertThrows(IllegalArgumentException.class, () -> {
+      new DisplayGpsReceiverImplementation(null);
+    });
+  }
+
+  @Test
+  @DisplayName ("Gps Signal Strength Signal Is Not Null")
+  public void measureGpsSignalStrengthSatelliteSignalIsNotNull() {
+    DisplayGpsReceiverImplementation d = 
+        new DisplayGpsReceiverImplementation(new DisplayGpsManagerStub());
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java]: Just a little cleaner formatting:
```suggestion
    DisplayGpsReceiverImplementation d = new DisplayGpsReceiverImplementation(
        new DisplayGpsManagerStub()
    );
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    assertThrows(IllegalArgumentException.class, () -> {
+      d.measureGpsSignalStrength(null);
+    });
+  }
+
+  @Test
+  @DisplayName ("Satellite Signal Is Not Null")
+  public void measureSignalSatelliteSignalIsNotNull() {
+    DisplayGpsReceiverImplementation d = 
+        new DisplayGpsReceiverImplementation(new DisplayGpsManagerStub());
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java]: Just a little cleaner formatting:
```suggestion
    DisplayGpsReceiverImplementation d = new DisplayGpsReceiverImplementation(
        new DisplayGpsManagerStub()
    );
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    assertThrows(IllegalArgumentException.class, () -> {
+      d.measureSignal(null);
+    });
+  }
+
+  @Test
+  @DisplayName ("Gps Signal Strength In Is Signal Strength Out ")
+  public void measureGpsSignalStrengthSatelliteInIsSignalOut() {
+    DisplayGpsReceiverImplementation d = 
+        new DisplayGpsReceiverImplementation(new DisplayGpsManagerStub());
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java]: Just a little cleaner formatting:
```suggestion
    DisplayGpsReceiverImplementation d = new DisplayGpsReceiverImplementation(
        new DisplayGpsManagerStub()
    );
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    Satellite s = new Satellite("New Satellite", 7);
+    assertEquals(s.getSatelliteName(), d.measureGpsSignalStrength(s));
+  }
+
+  @Test
+  @DisplayName ("Measure Signal In Is Signal Out")
+  public void measureSignalSignalInIsSignalOut() {
+    DisplayGpsReceiverImplementation d = 
+        new DisplayGpsReceiverImplementation(new DisplayGpsManagerStub());
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsReceiverImplementationTests.java]: Just a little cleaner formatting:
```suggestion
    DisplayGpsReceiverImplementation d = new DisplayGpsReceiverImplementation(
        new DisplayGpsManagerStub()
    );
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Few minor comments, mostly around formatting, then you're all set to merge!

**Review by andrewdimmer — APPROVED** [review]
Looks good to me, you're all set to merge provided checks pass!

**Pull request merged**